Objc-deliTests/FISAppDelegateSpec.m

```diff
@@ -15,58 +15,65 @@
     });
     
     beforeEach(^{
-        deliLine = [[NSMutableArray alloc] init];
-    });
-
-    describe(@"takeaNumberWithDeliLine:Name:", ^{
-        it(@"Should respond to the correct selector", ^{
-            expect(delegate).to.respondTo(@selector(takeANumberWithDeliLine:Name:));
-        });
-
-        it(@"Should return a deli with another person", ^{
-            expect([delegate takeANumberWithDeliLine:deliLine Name:@"Ada"]).to.equal(@[@"Ada"]);
-        });
+        appDelegate = [[FISAppDelegate alloc] init];
+        deliLine = [[NSMutableArray alloc] initWithArray:@[@"Anita", @"Alan", @"Ada", @"Aaron", @"Alan"] ];
+        deliString = [appDelegate stringWithDeliLine:deliLine];
     });
-
-    describe(@"nowServingWithDeliLine:", ^{
-        it(@"Should respond to the correct selector", ^{
-            expect(delegate).to.respondTo(@selector(nowServingWithDeliLine:));
+    
+    describe(@"stringWithDeliLine:", ^{
+        it(@"returns an NSString object", ^{
+            expect(deliString).to.beKindOf([NSString class]);
         });
-
-        it(@"Should remove the person from the deli", ^{
-            [deliLine addObject:@"Ada"];
-            [deliLine addObject:@"Al"];
-            expect([delegate nowServingWithDeliLine:deliLine]).to.equal(@[@"Al"]);
+        
+        it(@"returns the customers in line as a string", ^{
+            expect(deliString).to.equal(@"The line is:\n1. Anita\n2. Alan\n3. Ada\n4. Aaron\n5. Alan");
         });
-
-        it(@"Should return empty array for empty deli", ^{
-            expect([delegate nowServingWithDeliLine:deliLine]).to.equal(@[]);
+        
+        it(@"explains if the line is empty", ^{
+            [deliLine removeAllObjects];
+            deliString = [appDelegate stringWithDeliLine:deliLine];
+            expect(deliString).to.equal(@"The line is currently empty.");
         });
     });
 
-    describe(@"deliLine:", ^{
-        it(@"Should respond to the correct selector", ^{
-            expect(delegate).to.respondTo(@selector(deliLine:));
+    describe(@"addName:toDeliLine:", ^{
+        it(@"should insert a name at the end of the line", ^{
+            deliLine = [appDelegate addName:@"Michael" toDeliLine:deliLine];
+            expect(deliLine).to.equal(@[@"Anita", @"Alan", @"Ada", @"Aaron", @"Alan", @"Michael"]);
+            
+            deliLine = [appDelegate addName:@"Grace" toDeliLine:deliLine];
+            expect(deliLine).to.equal(@[@"Anita", @"Alan", @"Ada", @"Aaron", @"Alan", @"Michael", @"Grace"]);
         });
+    });
 
-        it(@"Should return the appropriate line", ^{
-            [deliLine addObject:@"Ada"];
-            [deliLine addObject:@"Al"];
-            expect([delegate deliLine:deliLine]).to.equal(@"The line is currently: 1. Ada 2. Al");
+    describe(@"serveNextCustomerInDeliLine:", ^{
+        it(@"should remove only the first name in the line", ^{
+            NSString *nextCustomer = [appDelegate serveNextCustomerInDeliLine:deliLine];
```

Comment:
If you aren't going to use the nextCustomer variable, don't assign the output to that variable. Just let the return go out into the ether